File: src/gestorAplicación/operaciones/Operacion.java
package gestorAplicacion.operaciones;

import java.util.ArrayList;
import java.util.Date;
import gestorAplicacion.modelos.Producto;

public abstract class Operacion {

	private Date fecha;
	private ArrayList<Producto> productos;

	protected Operacion(Date fecha, ArrayList<Producto> productos) {
		this.fecha = fecha;
	}

	protected Operacion(Date fecha) {
		this.fecha = fecha;
	}

	public abstract String generarFactura();

	public Date getFecha() {
		return fecha;
	}

	public void setFecha(Date fecha) {
		this.fecha = fecha;
	}

	public ArrayList<Producto> getProductos() {
		return productos;
	}

	public void setProductos(Producto producto) {
		this.productos.add(producto);
	}

}

File: python/gestorAplicacion/operaciones/Operacion.py
from abc import ABC, abstractmethod


class Operacion(ABC):
    def __init__(self, fecha, productos):
        self._fecha = fecha
        self._productos = productos

    def getFecha(self):
        return self._fecha

    def setFecha(self, fecha):
        self._fecha = fecha

    def getProductos(self):
        return self._productos

    def setProductos(self, productos):
        self._productos = productos

    @abstractmethod
    def generarFactura(self):
        pass

    @abstractmethod
    def getTotal():
        pass
